src/sage/arith/misc.py

```diff
@@ -1842,76 +1865,29 @@ def gcd(a, b=None, **kwargs):
     """
     # Most common use case first:
     if b is not None:
-        try:
-            m = a.gcd
-        except (AttributeError, TypeError):
-            e = py_scalar_to_element(a)
-            if e is a:
-                raise TypeError("unable to call gcd with {!r}".format(a))
-            m = e.gcd
-
-        try:
-            return m(b, **kwargs)
-        except TypeError:
-            return m(py_scalar_to_element(b), **kwargs)
+        a = py_scalar_to_element(a)
+        b = py_scalar_to_element(b)
+        return a.gcd(b, **kwargs)
 
+    from sage.categories.objects import Objects
     from sage.structure.sequence import Sequence
     seq = Sequence(py_scalar_to_element(el) for el in a)
-    if seq.universe() is ZZ:
-        return GCD_list(seq)
-    else:
-        return __GCD_sequence(seq, **kwargs)
-
+    U = seq.universe()
 
-GCD = gcd
+    if len(seq) == 0:
+        return U(Integer(0))
```

Comment:
```suggestion
    if not seq:
        return U(ZZ.zero())
```
Both of these are (marginally) faster tests.